pkg/utils/httputil/key_auth.go

```diff
@@ -7,35 +7,35 @@ import (
 )
 
 type KeyAuthMiddleware struct {
-	next http.Handler
 	keys []string
 }
 
-func UseKeyAuth(keys []string, next http.Handler) *KeyAuthMiddleware {
+func UseKeyAuth(keys []string) *KeyAuthMiddleware {
```

Comment:
Now this function becomes a 'constructor' function of `KeyAuthMiddleware`. By convention, constructor functions for struct `FooBar` should be named `NewFooBar`.